### DISCUSSION TITLE
AnimationTree: Update parameter value automatically for Blendspace1D / Blendspace2D

### DISCUSSION THREAD

**Discussion by MatthiasBae** [godotengine/godot-proposals#9348] [2D]
### Describe the project you are working on
I am working on a TopDown 2D survival game where the characters might have a lot of animations based on the equipment, move direction, facing direction, injuries etc. 

### Describe the problem or limitation you are having in your project
The problem is that I have to maintain two different systems to change the current animation / state in the AnimationTree.

**Advanced Expressions in Transitions to other states**

<img width="400" alt="5904NVOhw6" src="https://github.com/godotengine/godot-proposals/assets/52577744/e65dd7da-491f-4195-9e07-c13ce3e28f21">

**Script to change the parameters for Blendspaces**

example:
<img width="832" alt="SCN2UM9hx5" src="https://github.com/godotengine/godot-proposals/assets/52577744/0dfdd835-c465-4db0-bfaa-aa4273862b96">

This makes it more complicated for complex AnimationTrees to change the animation as it will be changed by two different systems.
Also making changes to the AnimationTree results in updating every path which can be pretty tedious.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

As the Blendspace1D and Blendspace2D are also some kind of transitions it would be useful to also implement some kind of Advanced Expressions or you are able to reference a property of a script which the animator will check and update the blendposition to the properties value.
It would minimize the code the developer has to write and easier to maintain complex AnimationTrees because you do not have to maintain two different approaches / systems. Also updating the AnimationTree does not result in updating the paths in code.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

In Unity you are able to create variables in the Animator and assign them to BlendTrees. Depending on its value the Animator automatically changes the animation.

See the variable "Direction"
<img width="396" alt="msedge_07MCJyewng" src="https://github.com/godotengine/godot-proposals/assets/52577744/050c7525-b058-4e52-a8d3-d2693dbe0b1c">

I would suggest some similar feature. Maybe it is possible to use the class which already will be used for the advanced expressions and make a reference on its properties.

**Comment by keyschain** [comment]
I'd love to see something like this added as well, this would be great for 3D workflows too. The less editing in a script the better!
